src/goose/utils/session_file.py

```diff
@@ -37,3 +45,17 @@ def session_file_exists(session_files_directory: Path) -> bool:
     if not session_files_directory.exists():
         return False
     return any(list_session_files(session_files_directory))
+
+
+def save_latest_session(file_path: Path, messages: List[Message]) -> None:
+    with tempfile.NamedTemporaryFile("w", delete=False) as temp_file:
```

Comment:
using `tempfile` is to ensure atomic operation while writing file.  It prevents from incomplete writes, losing original file dataduring the unexpected interruption or concurrency update   

I guess the chance of unexpected scenario is small. However, the file could contain 100000 tokens (around 400KB). It would be safe to use `tempfile` to prevent from the unexpected behaviours